test/parallel_api/algorithm/alg.sorting/sort.pass.cpp

```diff
@@ -200,29 +252,110 @@ struct test_sort_with_compare
+        // Run tests for USM device memory
+        test_usm<sycl::usm::alloc::device>(exec, tmp_first, tmp_last, expected_first, expected_last, first, last, n,
+                                           compare);
+    }
+#endif // _PSTL_SYCL_TEST_USM
+#endif // TEST_DPCPP_BACKEND_PRESENT
+
+    template <typename Policy, typename InputIterator, typename OutputIterator, typename OutputIterator2, typename Size,
+              typename Compare>
+    typename ::std::enable_if<is_base_of_iterator_category<::std::random_access_iterator_tag, InputIterator>::value,
+                              void>::type
+    operator()(Policy&& exec, OutputIterator tmp_first, OutputIterator tmp_last, OutputIterator2 expected_first,
+               OutputIterator2 expected_last, InputIterator first, InputIterator last, Size n, Compare compare)
+    {
+        test(exec, tmp_first, tmp_last, expected_first, expected_last, first, last, n, compare);
+    }
+
     template <typename Policy, typename InputIterator, typename OutputIterator, typename OutputIterator2, typename Size,
               typename Compare>
     typename ::std::enable_if<!is_base_of_iterator_category<::std::random_access_iterator_tag, InputIterator>::value,
-                            void>::type
-    operator()(Policy&& /* exec */, OutputIterator /* tmp_first */, OutputIterator /* tmp_last */, OutputIterator2 /* expected_first */,
-               OutputIterator2 /* expected_last */, InputIterator /* first */, InputIterator /* last */, Size /* n */, Compare /* compare */)
+                              void>::type
+    operator()(Policy&& /* exec */, OutputIterator /* tmp_first */, OutputIterator /* tmp_last */,
+               OutputIterator2 /* expected_first */, OutputIterator2 /* expected_last */, InputIterator /* first */,
+               InputIterator /* last */, Size /* n */, Compare /* compare */)
     {
     }
 };
 
-template<typename T>
+template <typename T>
 struct test_sort_without_compare
 {
-    template <typename Policy, typename InputIterator, typename OutputIterator, typename OutputIterator2, typename Size>
+#if TEST_DPCPP_BACKEND_PRESENT
+#if _PSTL_SYCL_TEST_USM
+    template <sycl::usm::alloc alloc_type, typename Policy, typename InputIterator, typename OutputIterator,
+              typename OutputIterator2, typename Size>
     typename ::std::enable_if<is_base_of_iterator_category<::std::random_access_iterator_tag, InputIterator>::value &&
-                            can_use_default_less_operator<T>::value, void>::type
-    operator()(Policy&& exec, OutputIterator tmp_first, OutputIterator tmp_last, OutputIterator2 expected_first,
-               OutputIterator2 expected_last, InputIterator first, InputIterator /* last */, Size n)
+                                  can_use_default_less_operator<T>::value,
+                              void>::type
+    test_usm(Policy&& exec, OutputIterator tmp_first, OutputIterator tmp_last, OutputIterator2 expected_first,
```

Comment:
I'm afraid but I've already seen the same (up to 95%) function above.... Can we share the code?

Reply:
Now each of these functions contain 24 lines of code, one of these functions - with compare predicate, and other - without compare predicate.